com.unity.perception/Runtime/GroundTruth/Labelers/BoundingBoxLabeler.cs

```diff
@@ -108,8 +116,73 @@ void OnRenderedObjectInfosCalculated(int frameCount, NativeArray<RenderedObjectI
                     };
                 }
 
+                if (!CaptureOptions.useAsyncReadbackIfSupported && frameCount != Time.frameCount) 
+                    Debug.LogWarning("Not on current frame: " + frameCount + "(" + Time.frameCount + ")");
+
+                if (IsVisualizationEnabled()) 
+                {
+                    Visualize();
+                    ReportFrameCount(frameCount);
+                }
+                
                 asyncAnnotation.ReportValues(m_BoundingBoxValues);
             }
         }
+
+        /// <inheritdoc/>
+        protected override void SetupVisualizationPanel(GameObject panel)
+        {
+            var toggle  = GameObject.Instantiate(Resources.Load<GameObject>("GenericToggle"));
+            toggle.transform.SetParent(panel.transform);
+            toggle.GetComponentInChildren<Text>().text = "Bounding Boxes";
+            toggle.GetComponent<Toggle>().onValueChanged.AddListener(enabled => {
+                EnableVisualization(enabled);
+            });
+
+            visualizationHolder = GameObject.Instantiate(Resources.Load<GameObject>("BoundsHolder"));
+            visualizationHolder.transform.SetParent(panel.transform.parent, false);
+            
+            visualizationPanelCache = new Dictionary<string, RectTransform>();
+        }
+
+        void Visualize()
+        {
+            foreach (var box in m_BoundingBoxValues)
+            {
+                var rectTrans = GetVisualizationPanel(box.label_name);
```

Comment:
This appears to assume that each label will only ever have one bounding box, which is not the case.

Reply:
changed to use `GetVisualizationPanel(box.label_name + "_" + box.instance_id);`